packages/pds/src/repo/prepare.ts

```diff
@@ -342,8 +229,92 @@ function assertNoExplicitSlurs(rkey: string, record: RepoRecord) {
   }
 }
 
-// ensures two rkeys are not far apart
-function rkeyIsInWindow(rkey1: TID, rkey2: TID) {
-  const ms = Math.abs(rkey1.timestamp() - rkey2.timestamp()) / 1000
-  return ms < 10 * MINUTE
+type FoundBlobRef = {
+  ref: BlobRef
+  path: string[]
+}
+
+export const blobsForWrite = (
+  record: RepoRecord,
+  validate: boolean,
+): PreparedBlobRef[] => {
+  const refs = findBlobRefs(record)
+  const recordType =
+    typeof record['$type'] === 'string' ? record['$type'] : undefined
+
+  for (const ref of refs) {
+    if (check.is(ref.ref.original, untypedJsonBlobRef)) {
+      throw new InvalidRecordError(`Legacy blob ref at '${ref.path.join('/')}'`)
+    }
+  }
+
+  return refs.map(({ ref, path }) => ({
+    cid: ref.ref,
+    mimeType: ref.mimeType,
+    constraints:
+      validate && recordType
+        ? CONSTRAINTS[recordType]?.[path.join('/')] ?? {}
+        : {},
+  }))
+}
+
+export const findBlobRefs = (
+  val: LexValue,
+  path: string[] = [],
+  layer = 0,
+): FoundBlobRef[] => {
+  if (layer > 10) {
```

Comment:
Just a gut check that 10 layers is enough for e.g. the images within a record-with-media (counting in my head I suppose it's probably fine, close to 5?).  It just feels a little conservative, I'd be down to increase this a bit either way.

Reply:
Yup fair point, I can bump it up 👍